sdk/src/types/block/semantic.rs

```diff
@@ -17,91 +17,120 @@ use crate::types::block::{
 
 /// Errors related to ledger types.
 #[derive(Debug)]
-pub enum ConflictError {
+pub enum TxFailureError {
     /// Invalid conflict byte.
-    InvalidConflict(u8),
+    InvalidTxFailure(u8),
 }
 
-impl fmt::Display for ConflictError {
+impl fmt::Display for TxFailureError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Self::InvalidConflict(byte) => write!(f, "invalid conflict byte {byte}"),
+            Self::InvalidTxFailure(byte) => write!(f, "invalid tx failure byte {byte}"),
         }
     }
 }
 
-impl From<Infallible> for ConflictError {
+impl From<Infallible> for TxFailureError {
     fn from(err: Infallible) -> Self {
         match err {}
     }
 }
 
 #[cfg(feature = "std")]
-impl std::error::Error for ConflictError {}
+impl std::error::Error for TxFailureError {}
 
 /// Represents the different reasons why a transaction can conflict with the ledger state.
 #[repr(u8)]
 #[derive(Debug, Copy, Clone, Eq, PartialEq, packable::Packable)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-#[packable(unpack_error = ConflictError)]
-#[packable(tag_type = u8, with_error = ConflictError::InvalidConflict)]
-pub enum ConflictReason {
+#[packable(unpack_error = TxFailureError)]
+#[packable(tag_type = u8, with_error = TxFailureError::InvalidTxFailure)]
+pub enum TxFailureReason {
```

Comment:
Can we rename these types to their full length, like `TransactionFailureReason`?

Reply:
Sure